newsroom/news_api/news/types.py

```diff
@@ -0,0 +1,138 @@
+import functools
+from typing import Any, ClassVar
+from typing_extensions import Self
+from pydantic import Field, field_validator, model_validator
+from content_api.errors import BadParameterValueError
+from newsroom.search.types import BaseSearchRequestArgs, NewshubSearchRequest
+
+
+# set of fields that can be specified in the include_fields parameter
+default_allowed_include_fields = {
+    "type",
+    "urgency",
+    "priority",
+    "language",
+    "description_html",
+    "located",
+    "keywords",
+    "source",
+    "subject",
+    "place",
+    "wordcount",
+    "charcount",
+    "body_html",
+    "readtime",
+    "profile",
+    "service",
+    "genre",
+    "associations",
+}
+
+# set of fields that are allowed to be excluded in the exclude_fields parameter
+default_allowed_exclude_fields = {
+    "version",
+    "versioncreated",
+    "firstcreated",
+    "headline",
+    "byline",
+    "slugline",
+}
+
+
+class NewsApiSearchRequestArgs(BaseSearchRequestArgs):
+    # TODO-ASYNC: check if we need to raise exception on unknown fields
```

Comment:
This might be ideal, if a client requests a field that is not supported by the API, then we should raise an error